compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/RegExpLexer.java

```diff
@@ -164,62 +116,123 @@ private STToken readTokenInReTerm() {
         int nextChar = peek();
         switch (nextChar) {
             case LexerTerminals.BITWISE_XOR:
+                this.reader.advance();
```

Comment:
since we do `this.reader.advance();` in all the switch cases shall we do it before the switch case? Similar to readToken() in BallerinaLexer.java